packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -185,27 +185,20 @@ class FlutterPlugin implements Plugin<Project> {
 
         File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
         Properties plugins = readPropertiesIfExist(pluginsFile)
-
-        plugins.each { name, _ ->
-            def pluginProject = project.rootProject.findProject(":$name")
-            if (pluginProject != null) {
-                project.dependencies {
-                    if (project.getConfigurations().findByName("implementation")) {
-                        implementation pluginProject
-                    } else {
-                        compile pluginProject
+        // Plugins right now only have a notion of debug or release.
+        // Profile goes to release.
+        def Boolean isDebug = project.gradle.startParameter.taskRequests.any {it.args[0].contains("Debug")}
```

Comment:
This might be a good opportunity to comment `flutter.gradle` and/or decompose processing steps into well-named procedures.

Reply:
I'd rather tackle that in a separate PR that doesn't change any other functionality.